### ISSUE TITLE
v2.17.0 regression related to W0407: preferred-modules

### ISSUE THREAD

**Issue by ssbarnea** [pylint-dev/pylint#8453]
### Bug description

v2.17.0 release broke the W0407 rule by making the match trigger on partial string match.

```
# now failing due to 'py' being part of the imported name:
import copy
# some:
import pytest
```

This was working fine in v2.16.3 but now we need to disable preffered-modules for the moment.

### Configuration

```ini
[tool.pylint.IMPORTS]
preferred-modules = ["py:pathlib", "unittest:pytest"]
```

### Command used

```shell
-
```

### Pylint output

```shell
W0407: Prefer importing 'pathlib' instead of 'py' (preferred-module)
```

### Expected behavior

No error

### Pylint version

```shell
$ pylint --version
pylint 2.17.0
astroid 2.15.0
Python 3.11.2 (main, Feb 16 2023, 11:46:09) [Clang 14.0.0 (clang-1400.0.29.202)]
```

Related: https://github.com/PyCQA/pylint/issues/7957

**Comment by DanielNoord** [comment]
@d33bs Would you have time to look into this? It feels like we shouldn't use `in` but rather `==`.

**Comment by d33bs** [comment]
Thanks @DanielNoord, taking a look into this and will open a PR for a fix!

**Comment by ssbarnea** [comment]
I am afraid that the bug is not fixed, even in 3.0.0a6 because I tried to uncomment the rule skip on my project and still got those errors.

I am running pylint using pre-commit and version `v3.0.0a6`, so maybe we should reopen it.

**Comment by Pierre-Sassoulas** [comment]
Would you mind providing a reproducer @ssbarnea, please ?
